extensions/stats/plugin.cc

```diff
@@ -55,45 +121,66 @@ void clearTcpMetrics(::Wasm::Common::RequestInfo& request_info) {
 
 void PluginRootContext::initializeDimensions() {
   // Clean-up existing expressions.
-  // Potential perf optimization: re-use the existing expressions.
   cleanupExpressions();
 
   // Seed the common metric tags with the default set.
-  tags_ = IstioDimensions::defaultTags();
-
-  // Process the dimension overrides
-  for (const auto& metric : config_.metrics()) {
-    if (metric.dimensions().empty()) {
-      continue;
+  std::vector<MetricFactory> factories = DefaultMetrics();
+  Map<std::string, std::vector<MetricTag>> metric_tags;
```

Comment:
can string here be const ref?

Reply:
STD container don't like references, right? I think you cannot use a reference as key type. In any case, this is a temporary map as it gets destroyed on exit.

Reply:
ahh okk it just allows const but no ref.. thanks!